pkg/store/bucket.go

```diff
@@ -305,6 +345,49 @@ func NewBucketStore(
 	enablePostingsCompression bool,
 	postingOffsetsInMemSampling int,
 	enableSeriesResponseHints bool, // TODO(pracucci) Thanos 0.12 and below doesn't gracefully handle new fields in SeriesResponse. Drop this flag and always enable hints once we can drop backward compatibility.
+) (*BucketStore, error) {
+	bucketStoreOpts := []BucketStoreOption{
+		WithChunksLimit(chunksLimiterFactory),
+		WithChunksSizeLimit(NewChunksLimiterFactory(noChunksSizeLimit)),
+	}
+	return New(logger,
+		reg,
+		bkt,
+		fetcher,
+		dir,
+		indexCache,
+		queryGate,
+		maxChunkPoolBytes,
+		debugLogging,
+		blockSyncConcurrency,
+		filterConfig,
+		enableCompatibilityLabel,
+		enablePostingsCompression,
+		postingOffsetsInMemSampling,
+		enableSeriesResponseHints,
+		bucketStoreOpts...,
+	)
+}
+
+// New creates a new bucket backed store that implements the store API against
+// an object store bucket. It is optimized to work against high latency backends.
+func New(
```

Comment:
I would then just rename this to newBucketStore if there is no need to have it exported as its only used in this package.

Reply:
Ah I see it now. Then I would suggest `NewBucketStoreWithOptions` and set a deprecation warning on the old method. Then slowly refactor. Once that is done, rename back to `NewBucketStore`.

Reply:
thanks!